src/analyzer.rs

```diff
@@ -1036,4 +1056,29 @@ type: object
         assert_eq!(&items.members[1].name, "iat");
         assert_eq!(&items.members[2].name, "id");
     }
+
+    #[test]
+    fn skipped_type_as_map_nested_in_array() {
+        init();
+        let schema_str = r#"
+properties:
+  records:
+    items:
+      additionalProperties:
+        type: string
+      type: object
+    type: array
+type: object
+"#;
+        let schema: JSONSchemaProps = serde_yaml::from_str(schema_str).unwrap();
+
+        let structs = analyze(schema, "Geoip").unwrap().0;
+
+        assert_eq!(structs.len(), 1);
+        assert_eq!(structs[0].members.len(), 1);
+        assert_eq!(
+            structs[0].members[0].type_,
+            "Option<Vec<BTreeMap<String, String>>>"
+        );
```

Comment:
i'd add an assert that the member name is `records` just for quick readability, but otherwise great tests!